ecc/bls12381/ff/fp.go

```diff
@@ -25,23 +25,30 @@ func (z *Fp) SetUint64(n uint64)       { z.toMont(&fpRaw{n}) }
 func (z *Fp) SetInt64(n int64)         { z.SetUint64(uint64(-n)); z.Neg() }
 func (z *Fp) SetOne()                  { z.SetUint64(1) }
 func (z *Fp) Random(r io.Reader) error { return randomInt(z.i[:], r, fpOrder[:]) }
-func (z Fp) IsZero() bool              { return z.IsEqual(&Fp{}) }
-func (z Fp) IsEqual(x *Fp) bool        { return ctUint64Eq(z.i[:], x.i[:]) == 1 }
+func (z Fp) IsZero() int               { return z.IsEqual(&Fp{}) }
+func (z Fp) IsEqual(x *Fp) int         { return ctUint64Eq(z.i[:], x.i[:]) }
 func (z *Fp) Neg()                     { fiatFpMontSub(&z.i, &fpMont{}, &z.i) }
 func (z *Fp) Add(x, y *Fp)             { fiatFpMontAdd(&z.i, &x.i, &y.i) }
 func (z *Fp) Sub(x, y *Fp)             { fiatFpMontSub(&z.i, &x.i, &y.i) }
 func (z *Fp) Mul(x, y *Fp)             { fiatFpMontMul(&z.i, &x.i, &y.i) }
 func (z *Fp) Sqr(x *Fp)                { fiatFpMontSquare(&z.i, &x.i) }
-func (z *Fp) Inv(x *Fp)                { z.expVarTime(x, fpOrderMinus2[:]) }
+func (z *Fp) Inv(x *Fp)                { z.ExpVarTime(x, fpOrderMinus2[:]) }
 func (z *Fp) toMont(in *fpRaw)         { fiatFpMontMul(&z.i, in, &fpRSquare) }
 func (z Fp) fromMont() (out fpRaw)     { fiatFpMontMul(&out, &z.i, &fpMont{1}); return }
+func (z Fp) Sgn0() int                 { zz := z.fromMont(); return int(zz[0] & 1) }
+func (z *Fp) CMov(x, y *Fp, b int) {
+	mask := 0 - uint64(b&0x1)
```

Comment:
We should document similar to https://pkg.go.dev/crypto/subtle#ConstantTimeSelect

Reply:
I added a string,  I will devote an entire PR only for documentation.